### ISSUE TITLE
clangd jump to .h file , and don't know where is the implementation of a function

### ISSUE THREAD

**Issue by bruce-the-terrible** [clangd/clangd#1775]
source code is neovim 
```
git clone https://github.com/neovim/neovim
cd neovim && make CMAKE_BUILD_TYPE=Release
```
editor is NVIM v0.9.2
Build type: Release
LuaJIT 2.1.0-beta3

a function named _lua_tolstring_ （and some other function like this) 
1、shortcut _gd_ jump to .h file 
2、shortcut _gr_ find all the references 
but all this two steps never goes to the implementation of this function

telescope.nvim also can't find the right position

It seems compile_commands.json  or cmakelists.txt  is not  fully parsed by neovim and clangd .

**Comment by HighCommander4** [comment]
Where in the project's source code is the definition of `lua_tolstring` that you would like clangd to navigate to?

**Comment by bruce-the-terrible** [comment]
Sorry guys

The implementation of lua_tolstring is not a part of neovim .  The C file named lj_api.c is  a part of .deps/ , which means external  package or library .

**Comment by HighCommander4** [comment]
Right, so `.deps/builds/src/luajit/src/lj_api.c` is not part of clangd's index, because it's not listed in the project's `compile_commands.json`. As a result, clangd does not know about this file and never opens or processes it.

**Comment by bruce-the-terrible** [comment]
I write a new issue in xmake project 
[](https://github.com/xmake-io/xmake/issues/4254)

Programmer  would read a source code package which can't be compiled . Make tools like xmake should provide a simple way to create a fake compile_commands.json ，which make neovim and lsp happy .
